source/common/stats/thread_local_store.h

```diff
@@ -498,6 +498,15 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
   NullHistogramImpl null_histogram_;
   NullTextReadoutImpl null_text_readout_;
 
+  Thread::ThreadSynchronizer sync_;
+  std::atomic<uint64_t> next_scope_id_{};
+  uint64_t next_histogram_id_ ABSL_GUARDED_BY(hist_mutex_) = 0;
+
+  StatNameSetPtr well_known_tags_;
+
+  mutable Thread::MutexBasicLockable hist_mutex_;
+  StatSet<ParentHistogramImpl> histogram_set_ ABSL_GUARDED_BY(hist_mutex_);
```

Comment:
I ran this fix by @jmarantz offline. It's covered now by running gcc fastbuild, which is not very satisfying, so maybe we can follow up with better testing somehow.